python/lsst/eo/pipe/biasShiftsTask.py

```diff
@@ -0,0 +1,286 @@
+    im_arr = readout_order_arr(exp[amp.getRawDataBBox()].image.array, amp)
+    
+    if averageMethod=="mean":
+        parallel_correction = np.mean(readout_order_arr(exp.image[amp.getRawParallelOverscanBBox()].array,amp), axis=0)
+        return np.mean(im_arr-parallel_correction, axis=1)
+   
+   
+def get_full_parallel_correction(exp, amp, parallel_skip=2):
+        
+    image = exp.image
+    
+    oscan_bbox = amp.getRawParallelOverscanBBox()
+    full_bbox = amp.getRawBBox()
+    
+    oscan_min_y = max(oscan_bbox.getMinY(), full_bbox.getMinY())
+    oscan_max_y = min(oscan_bbox.getMaxY(), full_bbox.getMaxY())
+    
+    parallel_oscan_bbox = lsst.geom.Box2I(minimum=lsst.geom.Point2I(full_bbox.getMinX(),oscan_min_y), maximum=lsst.geom.Point2I(full_bbox.getMaxX(),oscan_max_y))
+    parallel_correction_arr = np.mean(readout_order_arr(image[parallel_oscan_bbox].array[parallel_skip:],amp), axis=0)
+    return parallel_correction_arr
+
+
+def smooth_rows(oscan_rows, window=30, order=1, n_smooth_rows=150):
+    return smooth_rows_butter(oscan_rows, window, order, n_smooth_rows)
+
+def smooth_rows_butter(oscan_rows, window=30, order=1, n_smooth_rows=150):
+    b, a = scipy.signal.butter(1, 1/n_smooth_rows,btype='low')
+    return scipy.signal.filtfilt(b,a,oscan_rows,padlen=0)
+
+def smooth_rows_savgol(oscan_rows, window=30, order=1):
+    return scipy.signal.savgol_filter(oscan_rows, window, order)
+
+def smooth_rows_deriv(smoothed_rows, window=30, order=1):
+    return scipy.signal.savgol_filter(smoothed_rows, window,order,deriv=1,delta=1,mode='mirror')
+
+def find_mask_runs(mask):
+    return numpy.flatnonzero(np.diff(numpy.r_[numpy.int8(0), mask.view(numpy.int8), numpy.int8(0)])).reshape(-1, 2)
+
+def shift_in_rows(oscan_rows, window=30, order=1, threshold=1, n_smooth_rows=150, skip_rows=30, show_plot=False):
+    smoothed_deriv = smooth_rows_deriv(smooth_rows(oscan_rows, window, order, n_smooth_rows), window, order)
+    norm_factor =  (n_smooth_rows*window)**0.5/np.std(butter_highpass_filter(oscan_rows, 1, window,order=order))
+    motion =np.abs(smoothed_deriv)*norm_factor
+    shift_mask = motion > threshold
+    shift_mask[:skip_rows] = False
+    
+    if show_plot:
+        plt.plot(motion)
+        plt.axhline(threshold)
+        plt.show()
+        plt.close()
+    
+    shift_regions = find_mask_runs(shift_mask)
+    shift_peaks = []
+    for region in shift_regions:
+        shift_peaks.append(region[0]+np.argmax(motion[region[0]:region[1]]))
+    if len(shift_regions)==0: return np.array([np.NaN]*4)
+    return np.concatenate((np.array([shift_peaks]),shift_regions.T)).T
+
+def flat_kernel(window=30):
+    kernel = np.concatenate([np.ones(window), -1*np.ones(window-1)])
+    return kernel/window
+
+def odd_local_kernel(window=30):
+    kernel = np.concatenate([np.arange(window), np.arange(-window+1,0)])
+    kernel = kernel/np.sum(kernel[:window])
+    return kernel
+
+def shift_kernel(window=30):
```

Comment:
Why not put the choice of kernel implementation in the task configuration?   One really shouldn't have to change the code by hand to use a different implementation like `flat_kernel`.  If that's not the intent, then please delete `flat_kernel` and simplify `shift_kernel` and `odd_local_kernel` so that you just call the one function directly.